Source/TextExperiment/Component/ASTextLayout.mm

```diff
@@ -2385,7 +2385,7 @@ static void ASTextDrawRun(ASTextLine *line, CTRunRef run, CGContextRef context,
               if (mode) { // CJK glyph, need rotated
                 CGFloat ofs = (ascent - descent) * 0.5;
                 CGFloat w = glyphAdvances[g].width * 0.5;
-                CGFloat x = x = line.position.x + verticalOffset + glyphPositions[g].y + (ofs - w);
```

Comment:
This seems good. small diffs are terrific. Would you mind splitting this up? Thanks.